host/tests/classes/hdmiCEC.py

```diff
@@ -0,0 +1,358 @@
+    def getLogicalAddress(self):
+        """
+        Get the logical address of the device.
+
+        Args:
+            None.
+
+        Returns:
+            str: Logical address of the device. Value ranges from '0' - 'f'
+        """
+        result = self.utMenu.select( self.testSuite, "Get Logical Address")
+        connectionStatusPattern = r"Result HdmiCecGetLogicalAddress\(IN:handle:\[0x[0-9A-F]+\], OUT:logicalAddress:\[([0-9A-Fa-f]+)\]\)"
+        logicalAddress = self.searchPattern(result, connectionStatusPattern)
+
+        return logicalAddress
+
+    def getPhysicalAddress(self):
+        """
+        Get the physical address of the device.
+
+        Args:
+            None.
+
+        Returns:
+            str: Physical Address of the DUT.
+        """
+        result = self.utMenu.select( self.testSuite, "Get Phyiscal Address")
+
+        addressBytes = None
+        pattern = r"Result HdmiCecGetPhysicalAddress\(IN:handle:.*OUT:physicalAddress:\[([\da-fA-F]\.[\da-fA-F]\.[\da-fA-F]\.[\da-fA-F])\]"
+        physicalAddress = self.searchPattern(result, pattern)
+        if physicalAddress:
+            parts = physicalAddress.split(".")
+            high_byte = hex(int(parts[0], 16) << 4 | int(parts[1], 16))
+            low_byte = hex(int(parts[2], 16) << 4 | int(parts[3], 16))
+            addressBytes = [f"0x{int(high_byte, 16):02X}",f"0x{int(low_byte, 16):02X}"]
+
+        return addressBytes
+
+    def cecTransmitCmd(self, destLogicalAddress:str, cecCommand:str, cecData:list=None):
+        """
+        Transmit or broadcast a CEC command to a specified destination.
+
+        Args:
+            destLogicalAddress (str): Destination logical address.
+            cecCommand (str): CEC command in hexadecimal.
+            cecData (list, optional): List of data bytes to include in the transmission.
+
+        Returns:
+            None
+        """
+
+        promptWithAnswers = [
+                {
+                    "query_type": "direct",
+                    "query": "Enter a valid Destination Logical Address:",
+                    "input": destLogicalAddress
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Enter CEC Command (in hex):",
+                    "input": cecCommand
+                },
+        ]
+
+        if cecData:
+            for byte in cecData:
+               promptWithAnswers.append(
+                   {
+                       "query_type": "direct",
+                       "query": "Enter Databyte",
+                       "input": str(byte)
+                   })
+
+        result = self.utMenu.select( self.testSuite, "Transmit CEC Command",promptWithAnswers)
+
+        pattern = r"Result HdmiCecTx\(.*OUT:result:\[(.*?)\]\).*HDMI_CEC_STATUS:\[(.*?)\]"
+
+        txStatus = self.searchPattern(result, pattern)
+
+        status = False
+        if txStatus == "HDMI_CEC_IO_SENT_AND_ACKD" or txStatus == "HDMI_CEC_IO_SENT_BUT_NOT_ACKD":
```

Comment:
Should check for HDMI_CEC_IO_SENT_AND_ACKD when destination logical address is other than 0x0f (broadcast) for broadcast, it should be HDMI_CEC_IO_SENT_BUT_NOT_ACKD.  Should not check for either one of them.

Reply:
changed the condition check